csrc/scheduler/matmul_utils.cpp

```diff
@@ -294,6 +315,98 @@ std::string isMatmulFusionDefinitionSupported(
   return "";
 }
 
+// Assume that tens has a contiguous dimension, and that we will load rows of
+// that dimension, without merging with another dimension first. Then determine
+// the maximum vectorization that can be used.
+//
+// If the argument has no contiguous dimensions, then a vectorization width of 1
+// is returned.
+//
+// These rows can start at any multiple of the non-contiguous strides, so we
+// seek the largest power of 2 that divides all those other dimensions (capped
+// to 16) as well as the data pointer.
+int64_t maxRowVectorization(const at::Tensor& tens) {
+  const int64_t data_ptr_int =
+      static_cast<int64_t>(reinterpret_cast<std::uintptr_t>(tens.data_ptr()));
```

Comment:
For some reason `tens.data_ptr()` is always 0 currently. I am looking into this since it means we can't detect misaligned inputs currently (see failing test)

Reply:
Actually that will require more involved changes. The issue is here: https://github.com/NVIDIA/Fuser/blob/2f8961a983dc1e136958f94a7658912a465cf57f/csrc/kernel_cache.cpp#L1613-L1614
That calls this piece of code which ensures that we do not hold tensors indefinitely that would reserve their data waste memory:https://github.com/NVIDIA/Fuser/blob/2f8961a983dc1e136958f94a7658912a465cf57f/csrc/kernel_cache.cpp#L34-L70
However, it also means the heuristic does not have information about pointer alignment, even though it is available prior to this. It is actually acknowledged as well since pointer alignment is used to create our fast path's cache key: https://github.com/NVIDIA/Fuser/blob/2f8961a983dc1e136958f94a7658912a465cf57f/csrc/kernel_cache.cpp#L314-L316
Changing that is probably beyond the scope of this PR, but it should be noted that inputs that are misaligned due to slicing might cause errors until we fix this.

Reply:
Until then I will disable the failing tests and post an issue to track this.